From: Simon McVittie <smcv@debian.org>
Date: Wed, 4 May 2022 10:51:36 +0100
Subject: tests: Treat some more layout tests as flaky

Forwarded: no
---
 tests/test-layout.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/test-layout.c b/tests/test-layout.c
index 0043fde..c47b245 100644
--- a/tests/test-layout.c
+++ b/tests/test-layout.c
@@ -104,6 +104,12 @@ test_layout (gconstpointer d)
         }
       else
 #endif
+      if ((strstr (filename, "vertical.layout") || strstr (filename, "valid-20.layout"))
+          && g_getenv ("DEB_ALLOW_FLAKY_TESTS") == NULL)
+        {
+          g_test_incomplete ("layout calculated differently with different Harfbuzz versions");
+        }
+      else
         {
           g_test_fail ();
         }
